Добавлена функциональность "Отзывы"

H1ppalectry0n opened the pull request:
В приложении должны появиться отзывы на фильмы. Добавленные отзывы должны иметь рейтинг и несколько дополнительных характеристик.

H1ppalectry0n committed:
Добавлена функциональность "Отзывы"

В приложении должны появиться отзывы на фильмы. Добавленные отзывы должны иметь рейтинг и несколько дополнительных характеристик.

H1ppalectry0n committed:
Изменена настройка базы данных при тестировании

H1ppalectry0n committed:
Рефакторинг codestyle

carbonbasedlifeforms commented on src/main/resources/data.sql:
```diff
@@ -68,3 +68,16 @@ INSERT INTO film_likes (film_id, user_id)
 (2, 3);
+
+-- Пример отзыва
+INSERT INTO reviews (content, is_positive, user_id, film_id)
```
delete или truncate по этим таблицам не стал делать? не будет дублей при накате скрипта?

carbonbasedlifeforms commented on src/main/resources/application.properties:
```diff
@@ -2,7 +2,8 @@ logging.level.org.springframework.validation=TRACE
 spring.sql.init.mode=always
-spring.datasource.url=jdbc:h2:file:./db/filmorate
+#spring.datasource.url=jdbc:h2:file:./db/filmorate
+spring.datasource.url=jdbc:h2:mem:filmorate
```
почему решил поменять эту настройку? )

H1ppalectry0n replied:
это как раз к предыдущему вопросу. С этой настройкой база данных храниться в оперативной памяти и очищается при завершении приложения. Так удобней как по мне.

carbonbasedlifeforms commented on src/main/java/ru/yandex/practicum/filmorate/controller/ReviewController.java:
```diff
@@ -0,0 +1,70 @@
+    @GetMapping
+    public List<Review> getReviews(
+            @RequestParam(required = false) Long filmId,
+            @RequestParam(defaultValue = "10") long count
```
меня мой ревьювер постоянно просит выносить такое в константы ) но тут не обязательно, вкусовщина так сказать )

carbonbasedlifeforms commented on src/main/java/ru/yandex/practicum/filmorate/storage/ReviewStorage.java:
```diff
@@ -0,0 +1,136 @@
+        return jdbc.query(FIND_TOP_BY_FILM_ID, mapper, filmId, limit);
+    }
+
+    public Optional<Review> findById(Long id) {
```
есть по моим наблюдениям какая то бесконечная битва между сторонниками возвращать Опшнл и противниками - я на нейтральной стороне )

carbonbasedlifeforms commented on src/main/java/ru/yandex/practicum/filmorate/service/ReviewService.java:
```diff
@@ -0,0 +1,140 @@
+import java.util.List;
+
+@Service
+@Slf4j
```
аннотацию добавил, но почти не используешь (не логгируешь) тут )

ValeriiLindenPy commented on src/main/java/ru/yandex/practicum/filmorate/controller/ReviewController.java:
```diff
@@ -0,0 +1,70 @@
+    @GetMapping
+    public List<Review> getReviews(
+            @RequestParam(required = false) Long filmId,
+            @RequestParam(defaultValue = "10") long count
```
Думаешь Long не черезчур ? У нас просто в методе getTop() RequestParam int.

H1ppalectry0n replied:
Ну чисто технически `id` в базе данных имеют тип `BIGINT`, что соответствует `Long` в java. Соответственно количество записей, которые теоретически можно получить из базы, может превышать диапазон `Integer`.

ValeriiLindenPy commented on src/main/java/ru/yandex/practicum/filmorate/service/ReviewService.java:
```diff
@@ -0,0 +1,140 @@
+            throw new NotFoundException("Film with id = %d not found".formatted(reviewDto.getFilmId()));
+        }
+
+        if (reviewDto.getContent() != null) {
+            review.setContent(reviewDto.getContent());
+        }
```
`@NotBlank
    private String content;  `
Не лишняя ли проверка если он и так валидируется на Blanck?

H1ppalectry0n replied:
Да здесь логика потерялась. В dto `content` может быть null. Надо переработать.

H1ppalectry0n replied:
Исправил

ValeriiLindenPy commented on src/main/resources/application.properties:
```diff
@@ -2,7 +2,8 @@ logging.level.org.springframework.validation=TRACE
 
 spring.sql.init.mode=always
-spring.datasource.url=jdbc:h2:file:./db/filmorate
+#spring.datasource.url=jdbc:h2:file:./db/filmorate
```
Я делал хранение в файле , чтобы подключаться к ней через DBeaver. Но думаю это опциаонально действительно.

H1ppalectry0n replied:
И у тебя он подключался??
У меня нет в нем плагина для третьей версии h2

H1ppalectry0n committed:
Исправления после ревью

Базаданных снова пишеться в файл
Добавлена очиска созданных таблиц

Исправлена валидация класса RequestUpdateReviewDto

Исправлен обработчик ошибок валидации
